pkg/trace/agent/agent.go

```diff
@@ -259,6 +263,91 @@ func (a *Agent) Process(p *api.Payload, sublayerCalculator *stats.SublayerCalcul
 	}
 }
 
+var _ api.StatsProcessor = (*Agent)(nil)
+
+// ProcessStats processes incoming client stats in from the given language lang.
+func (a *Agent) ProcessStats(in pb.ClientStatsPayload, lang string) {
+	if in.Env == "" {
+		in.Env = a.conf.DefaultEnv
+	}
+	in.Env = traceutil.NormalizeTag(in.Env)
+	out := stats.Payload{
+		HostName: in.Hostname,
+		Env:      in.Env,
+	}
+	var buf strings.Builder
+	for _, group := range in.Stats {
+		for _, b := range group.Stats {
+			normalizeStatsGroup(&b, lang)
+			a.obfuscator.ObfuscateStatsGroup(&b)
+			a.Replacer.ReplaceStatsGroup(&b)
+
+			tags := map[string]string{"version": in.Version}
+			if b.HTTPStatusCode != 0 {
+				tags["http.status_code"] = strconv.Itoa(int(b.HTTPStatusCode))
+			}
+			newb := stats.Bucket{
+				Start:            int64(group.Start),
+				Duration:         int64(group.Duration),
+				Counts:           make(map[string]stats.Count),
+				Distributions:    make(map[string]stats.Distribution),
+				ErrDistributions: make(map[string]stats.Distribution),
+			}
+			grain, tagset := stats.AssembleGrain(&buf, out.Env, b.Resource, b.Service, tags)
+			key := stats.GrainKey(b.Name, stats.HITS, grain)
+			newb.Counts[key] = stats.Count{
+				Key:      key,
+				Name:     b.Name,
+				Measure:  stats.HITS,
+				TagSet:   tagset,
+				TopLevel: float64(b.Hits),
```

Comment:
@piochelepiotr can we actually not set TopLevel at all?

Reply:
Yes, we can skip it.

Reply:
Should I remove this completely or do it in another PR? Not sure what you mean here...

Reply:
You can remove it entirely on this PR (scoped to the new stat payload). This field is not used anymore

Reply:
Done.